routes/infrastructure/construct.py

```diff
@@ -53,36 +55,57 @@ def __init__(
                     origin=origins.HttpOrigin(
                         s3Bucket.bucket_website_domain_name,
                         protocol_policy=cf.OriginProtocolPolicy.HTTP_ONLY,
+                        origin_id="stac-browser",
                     ),
                     cache_policy=cf.CachePolicy.CACHING_DISABLED,
                 ),
                 certificate=domain_cert,
-                domain_names=[veda_route_settings.domain_hosted_zone_name]
+                domain_names=[f"{stage}.{veda_route_settings.domain_hosted_zone_name}"]
                 if veda_route_settings.domain_hosted_zone_name
                 else None,
                 additional_behaviors={
                     "/api/stac*": cf.BehaviorOptions(
```

Comment:
I missed this additional behaviors in the last PR, maybe we can drop a TODO here to make this configurable in the condition that a path prefex is used like `f"{veda_route_settings.stac_path_prefix}*": cf.BehaviorOptions(...`. I think we can work on that after using things for a while though, to see what our needs are